### PR TITLE
Lint & Format

### PR DESCRIPTION
**Pull request by irajagop**
- Added linting and formatting check with github actions
- Formatted entire codebase
- Fixed linter errors
- Removed `# noqa` with fix

### CODE REVIEW

**Comment by vbaddi** [comment]
cc @quic-jhugo we are planning on adding `github workflows` for the `linter checks`. Please review

File: .github/workflows/lint-format.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: pip3 install ruff
+      - run: ruff check
+    env:
+      RUFF_OUTPUT_FORMAT: github
+  format:
```
Comment (ochougul) [inline .github/workflows/lint-format.yml:12]: Does this format the code and push the formatted code to the branch?

Comment (irajagop) [reply .github/workflows/lint-format.yml:12]: This only raises error if the code is not of correct format and has linting errors.

**Review by vbaddi — APPROVED** [review]
👍

**Comment by quic-jhugo** [comment]
This seems fine to me

**Review by ochougul — APPROVED** [review]

**Pull request merged**